lib/ansible/modules/cloud/ovirt/ovirt_vm.py

```diff
@@ -1322,8 +1332,8 @@ def build_entity(self):
             stateless=self.param('stateless') or self.param('use_latest_template_version'),
             delete_protected=self.param('delete_protected'),
             bios=(
-                otypes.Bios(boot_menu=otypes.BootMenu(enabled=self.param('boot_menu')))
-            ) if self.param('boot_menu') is not None else None,
+                otypes.Bios(boot_menu=otypes.BootMenu(enabled=self.param('boot_menu')), type=otypes.BiosType(self.param('bios_type')))
+            ) if self.param('boot_menu') is not None or self.param('bios_type') is not None else None,
```

Comment:
the logic of `self.param('boot_menu') is not None` is always true because this PR changes its default to false. Changing the default of boot_menu seems unnecessary for this logic. Or did I miss anything?

Reply:
I set a default boot_menu for the case when only the bios_type parameter is provided. However there is a consequence to this that I have just discovered. 

If you provision a machine say with bios_type or boot_menu set, and then you subsequently leave them out of a task, e.g to 'suspend' a machine say, then the machine config is marked as changed because they revert to the default values.

I need to spend another afternoon looking through the 'marked as changed' logic. At the moment if you set a boot_menu or bios_type option you have to include that in all subsequent tasks that use the ovirt_vm module.